### ISSUE TITLE
Sprint - Nov 11 to Nov 22

### ISSUE THREAD

**Issue by marandaneto** [PostHog/posthog#26021]
# Global Sprint Planning

## 3 things that might take us down

1.
2.
3.

## Urgent incident follow-ups

https://github.com/orgs/PostHog/projects/103/views/2

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
# Team ___

**Support hero:** ___

## Retro

<!-- Grab the high and low priority items from last time and add whether that item was completed or not -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red


### High priority

-

### Low priority / side quests

-

```

**Comment by mariusandra** [comment]
# Team Crabs Ducks Pigeons

- Support: Ben

## Retro

- @mariusandra Get messaging v0 ready for dogfooding. 
   - ⚪ out for testing, but see below (on hold)
- @benjackwhite CDP UI/UX + coming back from holiday (odds/ends)
   - 🔴 catching up took time, no progress on CDP, all for next week
- @oliverb123 Olly 80% on error tracking. Improve live debugging of CDP.
   - 🟢 merged PR for better error handling
   - 🟢 error tracking --> being built
- @MarconLP Support sprint + new destinations
   - 🟢 coming along, being built

## OKRs
* Widespread usage of the new CDP 
* Messaging V1 
* Ingestion is rock solid

## Sprint focus

### High priority

- CDP and ads (whole team)

**Comment by pauldambra** [comment]
# Team Claire will leave me if I travel in December as well

Another 1 week shipping, 1 week travelling sprint 

**Support hero:** @richard-better with @pauldambra supporting     

## Retro



---

## Mobile

* mobile replay billing clarity @pauldambra 
* lots of small wins @ioannisj can see. Some potential issues:
  * https://github.com/PostHog/posthog-ios/issues/211
  * https://github.com/PostHog/posthog-ios/issues/210
  * https://github.com/PostHog/posthog-ios/issues/220
* refactor net logs instrumentation @ioannisj 
  * https://github.com/PostHog/posthog-ios/issues/204
* flutter replay alpha and getting feedback @marandaneto
    * no network logs for now
    
## Replay


* replay URL denylist @richard-better 
* a brief diversion before querying network data - wrap websockets as well as fetch @pauldambra  
    * this should include making it safer to roll out changes to recordings dependencies  
    
### Low priority / side quests

 * ✅ nobody is lost or arrested in Valencia
* trigger to start recordings on event @richard-better 
* lazy loading replay in the SDK @pauldambra 

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red

🟡📱Goal 1: People think of PostHog as a mobile solution
🟡 📹 Goal 2: Replay playback 3000
    - delayed-ish till Nov and Cory time in Valencia
🟡 🎤 Goal3: Ship the things users keep asking for
 
### High priority

## Mobile


## Replay


### Low priority / side quests

**Comment by Phanatic** [comment]
# Team Surveys

## Retro

✅  Beta testing no-code experiments with customers
✅  Surveys bug fixes
✅  No-code experiemnts bug fixes
✅  Investigate adaptive sampling for collecting survey responses.
✅ Support queues were split
⌛ Feature preview release of no-code experiments
⌛ Documentation for no-code experiments.

### Side Quests 


---

## OKRs

1. Implement Enterprise-ready features for Surveys
2. Increase adoption of Surveys.

### High priority
* Adaptive sampling for collecting survey responses.
* Support no-code experiments.

**Comment by jurajmajerik** [comment]
# Team Experiments

## Retro
- 🟡 ElevenLabs - risk of churning. Scrambling to implement holdouts with Neil before tomorrow's meeting with them - will be done.

- 🟡 Continued on HogQL for Experiments (https://github.com/PostHog/posthog/issues/24914) - good progress, 90% done - backend migration complete now, just need to finish the frontend.

- 🟡 Review statistics with a contractor
	- going great, we have a shared Jupyter notebook where we're writing the new methodology. Making sure to understand everything to fully revamp docs once this is done.
	- focusing on 3 main metric types: funnel, count, numerical field value (e.g. revenue)

- 🟢 Support hero for the Feature Success queue - workload very manageable

## Sprint goals
- ElevenLabs is the current priority
	- based on the outcome of tomorrow's meeting, we'll see what else needs to be done right now
	- reusable metrics they've been requesting are on my list already - this is the next step after HogQL migration
- Continue HogQL rewrite
	- migrate UI behind a feature flag
- Review the methodology with a statistician
	- discuss possibly measuring difference between averages (currently not working correctly), comparison between test variants

**Comment by robbie-c** [comment]
# Team web analytics

**Support hero:** @robbie-c 

## Retro

Onboarded Mike, currently doing a hackathon

#### Tasks
🟠 Finish cookieless prototype
🟢 Onboard Mike / have a successful offsite
🟢 Finish up the v1 sessions table PR

## OKR

* Continue to improve performance
* Build a small amount of more advanced marketing features, to measure interest
* Cookieless tracking
* Improve the NextJS experience
* More page-level insights
* Improve the documentation, and link to it more from the app

### High priority
Mike
- Finish https://github.com/PostHog/posthog/issues/21195
- UX improvements to the graphs, https://github.com/PostHog/posthog/issues/26000 and https://github.com/PostHog/posthog/issues/25999
- Improve goals query UX, https://github.com/PostHog/posthog/issues/26004 and https://github.com/PostHog/posthog/issues/25978

Robbie
- Finish cookieless prototype https://github.com/PostHog/posthog/issues/25117
- Make a start on page reports / core web vitals


### Stretch goals
Mike
- https://github.com/PostHog/posthog/issues/26003

**Comment by fuziontech** [comment]
# Team 🧊 Data

## OKR Q2 2024

### Objective

**Goals for Q4:**

- Improve elasticity and flexibility of our data store by putting all our data in Iceberg
    - [Work with Altinity](https://docs.google.com/document/d/12icx71-_kV0aICP3igtOc0di_yrey_W82HzbadtNPBY/edit?usp=sharing) effectively to ship read path for Iceberg on ClickHouse - Brett Hoerner
    - Setup infrastructure to ship all of our data to Iceberg on S3 - James Greenhill
    - Shipping Query logs to Iceberg - James Greenhill
- Continue improving CH operational expertise
    - Upgrade to a later version of Clickhouse - James Greenhill
    - Capacity planning - James Greenhill
    - Automation - Daniel Escribano
    - Put some of the basic mitigation operations in runbooks - Daniel Escribano
- Schema management
    - Tool for schema migration (coordinator schemas) - Daniel Escribano
    - Tool for long running mutations - Daniel Escribano
- Continued investment in performance
    - Tooling for other teams to understand which queries are slow and why - Ted Kaemming
    - Investigate variability of queries - Ted Kaemming
    - Per-team limits on queries/query complexity (needs product work) - Ted Kaemming

## Board
https://github.com/orgs/PostHog/projects/85/views/2

## Retro

### 🔥 Incidents 🔥 
- [#inc-2024-11-03-eu-ingestion-lag](https://posthog.slack.com/archives/C07U6SJD80P)
- [#inc-2024-10-29-clickhouse-us-lag](https://posthog.slack.com/archives/C07UNQ7QA48)
- [#inc-2024-10-29-clickhouse-eu-lag](https://posthog.slack.com/archives/C07TY2N0USE)

### Last Sprint's Goals
- Continue 24.8 upgrade and test fix @fuziontech @tkaemming help ❌ ♻️
- Work with DW/BE for pulling fresh data from Iceberg table OR Kafka for Batch Exports @bretthoerner ⚠️ 
- Ongoing Delete tasks (support) @fuziontech @Daesgar ✅ 
- Plan work with DW to start writing Iceberg @fuziontech  ⚠️ 
- Nullable performance improvements @tkaemming ✅ 
- Query log to Iceberg @fuziontech ❌ 
- Setup TTL for events for 48 months to gp3 disk @fuziontech ❌ ♻️
- EU to single AZ in eu-central-1c @Daesgar ✅ 
- Snapshot querying of Iceberg tables PR to CH @bretthoerner @fuziontech ⚠️ 

📣 Most work has been re-prioritized mid week in order to support snapshot diffing and re-clustering US
Tasks impacted marked with ❌ ♻️

## This Sprint's Goals 

### TL;DR

- Continue 24.8 upgrade and test fix @fuziontech @tkaemming
- Ongoing Delete tasks (support) @fuziontech @Daesgar 
- Remove unused Materialized Columns @tkaemming 
- Reshard US into 8 shards 3 replicas w/ coordinators @Daesgar 
- Snapshot querying of Iceberg tables PR to CH @bretthoerner @fuziontech

**Comment by EDsCODE** [comment]
# Team Data Storehouse
### OKR Q4 2024

### Objective

**Query 3000**

* **Key Results:**
    * Autocomplete
	* Increase general BI experience/product https://github.com/PostHog/meta/issues/157
	* Declutter the data warehouse UI and make the features intuitive to find

**Data Modeling MVP**

* **Key Results:**
	* Infrastructure decided and implemented
	* Integrating external data with feature flags
	* External data everywhere in insights/persons/cohorts
	* Get billing team to use modeling in posthog for their invoices_with_annual table
 
### Retro
- [x] Using warehouse person properties in experiment metrics
- [ ] SQL editor @EDsCODE 
- [ ] release dashboard insight variables to everyone @Gilbert09 
- [ ] BigQuery source release to everyone @Gilbert09 
- [ ] Investigate reducing temporal-data-warehouse pod size @Gilbert09 
- [x] Resolving batch exports replication lag - some solution, working with team clickhouse @tomasfarias 
- [x] More onboarding tasks @rossgray 
  - [x] Release chargebee source
  - [x] Select series colors in BI
  - [ ] HogQL ambiguous query error when joining tables with the same field names
  - [x] BI breakdowns

### Priority

- [ ] Release dashboard insight variables @Gilbert09 
- [ ] BigQuery source release to everyone @Gilbert09 
- [ ] Reduce temporal-data-warehouse pod size (memory) @Gilbert09 
- [ ] SQL editor 3000 MVP @EDsCODE 
- [ ] SPMC Batch Exports @tomasfarias
- [ ] Splitting persons batch exports @tomasfarias
- [ ] HogQL ambiguous query error when joining tables with the same field names @rossgray
- [ ] One of: Backfill UI, test connection button, or sessions/groups batch exports @rossgray

**Comment by zlwaterfield** [comment]
ℹ️ _Add your planning details as a comment below. I'll combine them into a single comment for global engineering planning._

----

_Last sprint plan as a jumping-off point:_ https://github.com/PostHog/billing/issues/950

# Team Growth

## Retro
- @zlwaterfield 
	 - [x] Support week 1
	 - RBAC
	 	- [x] Finish RFC for billing and get feedback on it
	 	- [x] Start working on it - get at least 1 PR merged for it
	- [ ] Get the in-app credit card entry experiment shipped 
	- [ ] Make a few improvements to the technical exercise (based on using it for 2 weeks)
	- [x] Project limit support requests (WIP)
		- https://posthog.slack.com/archives/C0428JHABK2/p1729272091461489 
		- https://posthoghelp.zendesk.com/agent/tickets/19005 
- @raquelmsmith 
  - [ ] Slowly roll out personless default change, deal with any issues - holding off for now
  - [ ] Fix bugs with product analytics templates onboarding flow
  - [x] Ship session replay "templates" screen
  - [x] Write data warehouse activation query
  - [x] Think about how to redo onboarding so people can pick multiple products at once
  - [x] Respond to customer about projects-on-invites
- @patricio-posthog 
	- [x] Billing limit: Get this bug fixed once and for all so we can move on
	- [ ] Forecasting bug: Fix forecasting and if it's a good idea, reimplement it with a new version 
	- [ ] Improve current `billing` API endpoints (optimize or spit current ones into separate endpoints)
	- [ ] Check if the webhooks for invoices are working fine (check events at Posthog) / turn off the daily sync to test that.

## Q4 Goals
✅=finished 🟡=in progress 🔴=won't finish ⚪=not started
1. 🟡 Get to a 4/10 in Enterprise platform support (Zach)
	- 🟡 Release Enterprise addon and iterate with sales to find a pricing model that makes sense
	- 🟡 Build RBAC
2. 🟡 Make revenue & forecasting more reliable (Patricio)
	- 🟡 Rejig how we do sync_invoices so we don’t see big drops - respond to webhooks instead of large syncs
	- ⚪ Evaluate temporal vs celery for usage reports and billing tasks so we can adhere to rate limits better
	- ⚪ Side hustle, if time permits: Rejig how anonymous / identified events are billed, as the current implementation is confusing.
3. 🟡 Activation: reverse trials, analytics onboarding flow, understand CC activation (Raquel)
	- ⚪ Dig into the data, watch replays to figure out why CC activation rate has been falling
	- ⚪ Add reverse trial to onboarding
	- 🟡 Add reverse trial to billing page for teams

## This sprint

Time off: 

- @zlwaterfield 
	 - RBAC
	 	- [ ] Get access models shipped
	 	- [ ] Get base middleware implements
	 	- [ ] Pick first product and get the base RBAC setup for it
	- [x] Get Teams Trials fully live
	- [x] Ship lowering of self serve credits requirements
	- [ ] Finish and ship project limits API checks
	- if time: 
		- [ ] (from last time) Make a few improvements to the technical exercise (based on using it for 2 weeks) 
		- [ ] (from last time) Get the in-app credit card entry experiment shipped 
		- [ ] Review existing billing metrics, hookup to billing on call, see if we should add others
		- [ ] customer io transaction api
- @patricio-posthog 
	- [ ] Continue with forecasting bugs, trying to understand what's happening and how to solve it.
	- [ ] (From last time): Turn off daily invoice sync to rely 100% on webhooks.
	- [ ] Try to optimize `/billing` endpoint, maybe separating forecasting into its own endpoint (after fixing it and understand how it works)
	Extra things if there is time:
	- [ ] Start to analyze forecasting and see its accuracy after each period.
- @raquelmsmith 
  - [ ] Get product analytics dashboard templates out again
  - [ ] follow up on replay templates / reengagement campaign
  - [ ] Daniel in-person onboarding
  - [ ] Multi-product-select in onboarding (with Daniel)

**Comment by aspicer** [comment]
# Team Product Analytics

**Support hero:** @aspicer and @anirudhpillai 

## Retro

Sandy

*   [✅](https://emojipedia.org/check-mark-button) Support (7 days)
*   [🟢](https://emojipedia.org/large-green-circle) 24.8 query porting
*   [🟢](https://emojipedia.org/large-green-circle) Continue porting cohorts

Thomas (Returns on Nov 4th!)

*   [✅](https://emojipedia.org/check-mark-button) Support

Anirudh (OOO through Nov 6th)

*   [🟢](https://emojipedia.org/large-green-circle) Resume alerting roll-out

## Upcoming Sprint

Support - Anirudh (Nov 11 - 15), Sandy (Nov 18-22)  
  
Sandy (1 week support)

*   Investigate getting date first seen onto a person #11952 #3031 
*   Password protection (for third party shares) #10869

  
Anirudh (1 week support)

*   Currently rolled out to 20% of users, sprint goal to roll out to 100%
*   Add slack messaging alerts
*   RFC for Saved Filters / Saved HogQL Snippets #20314 #2895
*   (Side Quest) Prometheus push gateway for gathering metrics from Celery 

Thomas

*   Filter Removal (Dashboard templates)
*   #23093 - Look into colors
*   #25609 - Chart reloading when disable breakdowns

**Comment by Twixes** [comment]
# Team AI Product Manager

## Retro

### High priority

- 🟢 Funnels support @skoob13 
- 🟢🔁 Tune agent/prompt for more relevant results (primary: excluding most $-prefixed events/properties) @skoob13 @Twixes
- 🔴 Stream feedback on the agent's current task in conversation @Twixes
- 🔴 PostHog dashboard for AI product manager use @Twixes
- 🔴 Generalize Session Replay's OpenAI data processing opt-in (to be determined with @fraserhopper) @Twixes 

### Low priority / side quests

- 🔴 Results evaluation framework - a benchmark @skoob13
- 🟡 Environments out as a feature preview @Twixes

## Hang over items from previous sprint

- Generalize Session Replay's OpenAI data processing opt-in (to be determined with @fraserhopper) 
- Stream feedback on the agent's current task in conversation
- PostHog dashboard for AI product manager use
- Results evaluation framework - a benchmark

## Objective

LLM-based product manager as a service

### High priority

- Results summarization
- Generalize Session Replay's OpenAI data processing opt-in (to be determined with @fraserhopper) @Twixes
- Stream feedback on the agent's current task in conversation @Twixes
- PostHog dashboard for AI product manager use @Twixes
- Tune Trends generation for more relevant results (smarter application of breakdowns, better property value examples) @skoob13
- Handle errors (agent recursion limit + graceful handling of e.g. ClickHouse errors) @skoob13

### Low priority / side quests

- Results evaluation framework - a benchmark @skoob13
- Environments out as a feature preview @Twixes

**Comment by benjackwhite** [comment]
# Team Infrastructure

**Support hero:** @danielxnj 

## Retro

- 🟡 Finish ArgoCD migration @bciaraldi 
  - https://github.com/PostHog/charts/issues/1836  
  - Almost all off-the-shelf stuff is migrated
- 🟢 Finish spot instances (at least for Decide and events, which are most expensive and can handle random load the best) @xneyder 
  - Decide/events/monitoring all moved over
  - Tuned a bunch of limits and instance types to match a spot workload better
  - Final more sensitive services like recordings and ingestion need some more tweaking	
- 🔴 Reserve instances @xneyder 
  - Purposefully delayed until spot work is fully done as we are making big savings there already that affect the calculation
- 🔴 Move most web/workers to same AZ as Clickhouse + easy failover.
  - (Will follow up after)
- (Half the time taken up by onboarding/offsite)

## OKR

1. 🏃 Continuous Delivery
2. 💰 Cost visualization
3. 🧪 Continuous Integration

### High priority
- 100% do the reservation changes ASAP - @benjackwhite 
- Fine tune our instance types to match our deployment workloads (cost optimisation to utilise the most of a node) @danielxnj 
- Optimize the LB/envoy deployments to remove all cross AZ traffic (cost optimization) @MichaelKutsch-ph 
- Tune the Warpstream dedicated node pools @danielxnj 
- ArgoCD - move lowest priority posthog services over (cdp, temporal-workers, cymbal) @bciaraldi 
- _(Will potentially update some outcomes of the offsite later)_

### Low priority / side quests

- Docker cache (working with Tom as warehouse still uses minio) @danielxnj

**Comment by dmarticus** [comment]
# Team Flags

## Retro

Seeing some non-deterministic behavior from the maxmindDB; seems like it starts to return different values from the IP address we expect sometimes.  Still figuring out exactly what the resolution is here (see thread: https://posthog.slack.com/archives/C06GG249PR6/p1730842770193459), but wanted to raise here since it's a cross-team dependency.

### Things we got done

- [x]  Finish static and dynamic cohort matching for new feature flag service
    - [x] support full subset of cohort match operations (`not_in`, etc) at the service/DB level
- [x] Prepare for Haven's onboarding in Seattle

---

## OKRs

1. Ship new feature flags in production and accomplish the 10x scale + 2x speed boost performance targets (ideally decreases infra spend by N% as well, will try to figure out a more specific target going forward)
2. Maintain current /decide service and make optimizations.

## Sprint

### High priority

- [ ] Implement holdout group targeting in new `/flags` - @dmarticus 
- [ ] Port over existing feature flag integration tests from the Django to rust repo (lots of valuable alpha in our existing `/decide` tests, but this doesn't need to happen this sprint) - @dmarticus 
- [ ] - Continue to onboard and get up to speed - @havenbarnes
    - [ ]  Flags Django app (flags UI)
    - [ ]  Flags decide service (old django service, new Rust service)
- [ ] https://github.com/PostHog/posthog/issues/25214 - @havenbarnes
